File: go-leetcode/array/48.旋转图像.go
package leetcode

func rotateMatrix(matrix [][]int) {
	// 主要看旋转后的坐标变化 matrix[i][j] -> matrix[j][len - 1 - i]
	// 直接换会导致有些数据会被替换，需要创建新数组
	n := len(matrix)
	newMatrix := make([][]int, n)
	for i := 0; i < n; i++ {
		newMatrix[i] = make([]int, n)
	}

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			newMatrix[j][n-1-i] = matrix[i][j]
		}
	}

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			matrix[i][j] = newMatrix[i][j]
		}
	}
}

File: python-leetcode/array/48.旋转图像.py
from typing import List


class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        # 主要看旋转后的坐标变化 matrix[i][j] -> matrix[j][len - 1 - i]
        # 直接换会导致有些数据会被替换，需要创建新数组
        n = len(matrix)
        new_matrix = [[0] * n for _ in range(n)]

        for i in range(n):
            for j in range(n):
                new_matrix[j][n - 1 - i] = matrix[i][j]

        for i in range(n):
            for j in range(n):
                matrix[i][j] = new_matrix[i][j]
